src/Components/Modal.tsx

```diff
@@ -34,8 +36,15 @@ export const useModal = (children: ComponentChildren, _options?: ModalOptions) =
 
     const [shown, setShown] = useState(options.shownInitially);
 
-    const show = useCallback(() => setShown(true), []);
-    const dismiss = useCallback(() => setShown(false), []);
+    const show = useCallback(() => {
+        setShown(true);
+        options.onShow?.();
+    }, [options.onShow]);
```

Comment:
ESLint isn't happy about that:

![image](https://user-images.githubusercontent.com/4048809/166665476-b2f36856-5f01-42e7-84e4-2a5120fff605.png)

As far as I understand it, you either need to depend on `options` as a whole or destructure it before.

Reply:
Yes, but in the same file it accepts similar syntax for another property. I think this should be right, but maybe I shouldn't doubt ESlint.